### PR TITLE
Docs and tests for HW Monotonic counter

### PR DESCRIPTION
**Pull request by oyvindronningstad**
This tests and documents #1917, and is based on #1919

### CODE REVIEW

File: include/bl_stash.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+See :ref:`bootloader_provisioning` for more information about the provisioning data and how the bootloader uses it.
+
+There are tests for the library at :file:`tests/subsys/bootloader/bl_stash/`.
+
+Monotonic version counter
+*************************
+
+The provision page can contain a monotonic version counter for the app.
+This is implemented as a series of 16 bit values, where each update to the counter is written to the next available slot.
+When reading the counter, bl_provision iterates over each slot checking that each is larger than the previous, to guarantee that it is monotonic (always increases).
```
Comment (frkv) [inline include/bl_stash.rst]: Need decision point on strictly monotonic vs monotonic accepting any-value-higher-than
The decision needs to be documented to ensure clarity on what is meant by "monotonic"

Comment (oyvindronningstad) [reply include/bl_stash.rst]: Added as a discussion point on #1917

**Comment by b-gent** [comment]
Should I review the RST on this PR or are you planning a doc review later? @oyvindronningstad

**Comment by oyvindronningstad** [comment]
> Should I review the RST on this PR or are you planning a doc review later? @oyvindronningstad

@b-gent This PR is based on #1919 so it would be great if you could review that first.

**Comment by oyvindronningstad** [comment]
Merged back into #1917

**Pull request closed without merging**